src/Gridify/Builder/BaseQueryBuilder.cs

```diff
@@ -129,10 +138,69 @@ public TQuery Build(ExpressionSyntax expression)
 
       if (hasIndexer)
          query = AddIndexerNullCheck(mapTarget, query);
+      else
+         query = AddNullPropagator(mapTarget, query);
 
       return ((TQuery)query, false);
    }
 
+
+   private object AddNullPropagator(LambdaExpression mapTarget, object query)
+   {
+      if (!mapper.Configuration.AvoidNullReference)
+         return query;
+
+      var mainQuery = query as LambdaExpression;
+      if (mainQuery == null)
+         return query;
+
+      var nullPropagatedExpression = BaseQueryBuilder.MemberNullPropagatorVisitor.Visit(mapTarget) as LambdaExpression;
+      if (mainQuery.Body is MethodCallExpression methodExp
+         && nullPropagatedExpression!.Body.Type != typeof(object)
+         && methodExp.Arguments.Count == 1)
+      {
+         var call = Expression.Call(methodExp.Method!, nullPropagatedExpression!.Body);
+         return Expression.Lambda(call, mainQuery.Parameters);
+      }
+
+      var body = mainQuery.Body;
+      if (body is not BinaryExpression bExp)
+         return query;
+
+      if (bExp.Method == null
+         || nullPropagatedExpression!.Body.Type == typeof(object)
+         || nullPropagatedExpression.Body is MethodCallExpression)
+      {
+         return query;
+      }
+      var newExp = Expression.Call(bExp.Method!, nullPropagatedExpression.Body, bExp.Right);
+      return Expression.Lambda(newExp, mainQuery.Parameters);
+   }
+
+   private object AddNullPropagator2(LambdaExpression mapTarget, object query)
```

Comment:
This is almost duplicate implementation, would be nice if you clean it up and keep just one version